pkgs/servers/gotify/ui.nix

```diff
@@ -10,16 +10,52 @@ yarn2nix-moretea.mkYarnPackage rec {
 
   version = import ./version.nix;
 
-  src_all = fetchFromGitHub {
-    owner = "gotify";
-    repo = "server";
-    rev = "v${version}";
-    sha256 = import ./source-sha.nix;
+  src_all = applyPatches {
+    src = fetchFromGitHub {
+      owner = "gotify";
+      repo = "server";
+      rev = "v${version}";
+      sha256 = import ./source-sha.nix;
+    };
+    postPatch = ''
+      substituteInPlace ui/yarn.lock \
+        --replace \
+          "https://registry.npmjs.org/caniuse-lite/-/caniuse-lite-1.0.30001237.tgz" \
+          "https___registry.npmjs.org_caniuse_lite___caniuse_lite_1.0.30001237.tgz"
+    '';
   };
   src = "${src_all}/ui";
 
   buildPhase = ''
+    export HOME=$(mktemp -d)
+    export WRITABLE_NODE_MODULES="$(pwd)/tmp"
+    mkdir -p "$WRITABLE_NODE_MODULES"
+
+    # react-scripts requires a writable node_modules/.cache, so we have to copy the symlink's contents back
+    # into `node_modules/`.
+    # See https://github.com/facebook/create-react-app/issues/11263
```

Comment:
This comment is good but perhaps should be at the beginning of the `buildPhase`.

Reply:
There are a few more comments related to the blocks separated by whitespace, so I think it's OK to to leave it as-is, right? :)